lib/mix/tasks/dialyze.ex

```diff
@@ -82,6 +82,7 @@ defmodule Mix.Tasks.Dialyze do
     else
       warnings ->
         print_warnings(warnings)
+        if (match?([_|_], warnings)), do: System.halt(1)
```

Comment:
Can we raise a mix error instead, which includes the number of dialyer warnings?